statsmodels/base/model.py

```diff
@@ -443,8 +443,13 @@ def fit(self, start_params=None, method='newton', maxiter=100,
             Hinv = np.linalg.inv(-retvals['Hessian']) / nobs
         elif not skip_hessian:
             try:
-                Hinv = np.linalg.inv(-1 * self.hessian(xopt))
-            except:
+                H = -1 * self.hessian(xopt)
+                w, v = np.linalg.eigh(H)
```

Comment:
If we are already calculating the eigenvalues, then we should keep them.
I think we can add them to mlefit.`mle_retvals`, `retvals` see line 474 below